client/request.go

```diff
@@ -227,13 +227,8 @@ func (cli *Client) checkResponseErr(serverResp serverResponse) error {
 		return fmt.Errorf("request returned %s for API route and version %s, check if the server supports the requested API version", http.StatusText(serverResp.statusCode), serverResp.reqURL)
 	}
 
-	var ct string
-	if serverResp.header != nil {
-		ct = serverResp.header.Get("Content-Type")
-	}
-
 	var errorMessage string
-	if (cli.version == "" || versions.GreaterThan(cli.version, "1.23")) && ct == "application/json" {
+	if serverResp.header.Get("Content-Type") == "application/json" && (cli.version == "" || versions.GreaterThan(cli.version, "1.23")) {
```

Comment:
this change removes the nil check for `serverResp.header` before accessing it:
```
if serverResp.header != nil {
		ct = serverResp.header.Get("Content-Type")
	}
```
is this safe?

Reply:
Reading is.. writing isn't 😂 https://go.dev/play/p/l0UWcwQU7qT

```go
package main

import (
	"fmt"
	"net/http"
)

type serverResponse struct {
	header http.Header
}

func main() {
	sr := serverResponse{}
	fmt.Printf("hello is: %q\n", sr.header.Get("hello"))

	sr2 := serverResponse{header: nil}
	fmt.Printf("hello is: %q\n", sr2.header.Get("hello"))

	// this panics
	sr.header.Set("hello", "world")
}
```